plugins/plugin-hosted/codenvy-machine-hosted/src/test/java/com/codenvy/machine/backup/MachineBackupManagerTest.java

```diff
@@ -416,99 +410,122 @@ public void shouldBackupWithCleanupAfterFinishOfCurrentRestore() throws Exceptio
+                                              this::runRestore);
+
+        backupFreezer.unfreeze();
         awaitFinalization();
 
+        // then
         verify(backupManager, times(2)).execute(cmdCaptor.capture(), anyInt());
-
         assertEquals(cmdCaptor.getValue()[0], BACKUP_SCRIPT);
     }
 
-    @Test
-    public void shouldIgnoreAllBackupsAndRestoresWhileBackupWithCleanupInQueueAfterBackup() throws InterruptedException,
-                                                                                                   TimeoutException,
-                                                                                                   IOException {
-        injectWorkspaceLock(WORKSPACE_ID);
+    /**
+     * Runs processes to do specified task simultaneously and wait until they ends.
+     *
+     * @param task
+     *        specifies task which will be run
+     * @param times
+     *         number of parallel jobs with specified task
+     * @throws InterruptedException
+     */
+    private void executeTaskNTimesWithBarrier(Runnable task, int times) throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(times);
 
-        executeNewBackup();
-        sleep(FAKE_BACKUP_TIME_MS / 4);
-        executeNewBackupWithCleanup();
-        sleep(FAKE_BACKUP_TIME_MS / 4);
-        executeNewBackup();
-        executeNewRestore();
-        sleep(FAKE_BACKUP_TIME_MS / 4);
-        executeNewBackup();
-        executeNewRestore();
+        for (int i = 0; i < times; i++) {
+            executor.execute(() -> {
+                task.run();
+                latch.countDown();
+            });
+        }
 
-        awaitFinalization();
+        latch.await();
+    }
 
-        verify(backupManager, times(2)).execute(cmdCaptor.capture(), anyInt());
+    /**
+     * Runs given tasks simultaneously and wait until they ends.
+     *
+     * @param tasks
+     *        tasks to run
+     * @throws InterruptedException
+     */
+    private void executeTasksSimultaneouslyWithBarrier(Runnable ... tasks) throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(tasks.length);
 
-        assertEquals(cmdCaptor.getValue()[0], BACKUP_SCRIPT);
+        for (Runnable task : tasks) {
+            executor.execute(() -> {
+                task.run();
+                latch.countDown();
+            });
+        }
+
+        latch.await();
     }
 
-    private void executeNewBackup() {
-        executor.execute(() -> {
-            try {
-                backupManager.backupWorkspace(WORKSPACE_ID, SRC_PATH, SRC_ADDRESS);
-            } catch (ServerException e) {
-                LOG.error(e.getLocalizedMessage(), e);
-            }
-        });
+    private void runBackup() {
+        try {
+            backupManager.backupWorkspace(WORKSPACE_ID, SRC_PATH, SRC_ADDRESS);
+        } catch (ServerException e) {
+            LOG.error(e.getMessage());
```

Comment:
Yes, sometimes backup/restore can fail with exception, but all what we want to know here is whether backup/restore have been performed. Other tests checks error messages.